Mathlib/CategoryTheory/Adjunction/Mates.lean

```diff
@@ -73,199 +74,524 @@ This can be seen as a bijection of the 2-cells:
-      adj₁.left_triangle_components]
-    dsimp
-    simp only [id_comp, ← Functor.comp_map, ← Functor.comp_obj, NatTrans.naturality_assoc]
-    simp only [Functor.comp_obj, Functor.comp_map, ← Functor.map_comp]
-    have : Prefunctor.map L₁.toPrefunctor (NatTrans.app adj₁.unit X) ≫
-        NatTrans.app adj₁.counit (Prefunctor.obj L₁.toPrefunctor X) = 𝟙 _ := by simp
-    simp [this]
-  -- See library note [dsimp, simp].
-  right_inv h := by
-    ext X
-    dsimp
-    simp [-Functor.comp_map, ← Functor.comp_map H, Functor.comp_map R₁, -NatTrans.naturality, ←
-      h.naturality, -Functor.map_comp, ← Functor.map_comp_assoc G, R₂.map_comp]
-#align category_theory.transfer_nat_trans CategoryTheory.transferNatTrans
-
-theorem transferNatTrans_counit (f : G ⋙ L₂ ⟶ L₁ ⋙ H) (Y : D) :
-    L₂.map ((transferNatTrans adj₁ adj₂ f).app _) ≫ adj₂.counit.app _ =
-      f.app _ ≫ H.map (adj₁.counit.app Y) := by
-  erw [Functor.map_comp]
-  simp
-#align category_theory.transfer_nat_trans_counit CategoryTheory.transferNatTrans_counit
-
-theorem unit_transferNatTrans (f : G ⋙ L₂ ⟶ L₁ ⋙ H) (X : C) :
-    G.map (adj₁.unit.app X) ≫ (transferNatTrans adj₁ adj₂ f).app _ =
-      adj₂.unit.app _ ≫ R₂.map (f.app _) := by
-  dsimp [transferNatTrans]
-  rw [← adj₂.unit_naturality_assoc, ← R₂.map_comp, ← Functor.comp_map G L₂, f.naturality_assoc,
-    Functor.comp_map, ← H.map_comp]
-  dsimp; simp
-#align category_theory.unit_transfer_nat_trans CategoryTheory.unit_transferNatTrans
-
--- See library note [dsimp, simp]
-end Square
+@[simps]
+def Mates :
+    (G ⋙ L₂ ⟶ L₁ ⋙ H) ≃ (R₁ ⋙ G ⟶ H ⋙ R₂) where
+      toFun := fun α ↦
+        whiskerLeft (R₁ ⋙ G) adj₂.unit ≫
+        whiskerRight (whiskerLeft R₁ α) R₂ ≫
+        whiskerRight adj₁.counit (H ⋙ R₂)
+      invFun := fun β ↦
+        whiskerRight adj₁.unit (G ⋙ L₂) ≫
+        whiskerRight (whiskerLeft L₁ β) L₂ ≫
+        whiskerLeft (L₁ ⋙ H) adj₂.counit
+      left_inv := by
+        intro α
+        ext
+        unfold whiskerRight whiskerLeft
+        simp only [comp_obj, id_obj, Functor.comp_map, comp_app, map_comp, assoc, counit_naturality,
+          counit_naturality_assoc, left_triangle_components_assoc]
+        rw [← assoc, ← Functor.comp_map, α.naturality, Functor.comp_map, assoc, ← H.map_comp,
+          left_triangle_components, map_id]
+        simp only [comp_obj, comp_id]
+      right_inv := by
+        intro β
+        ext
+        unfold whiskerLeft whiskerRight
+        simp only [comp_obj, id_obj, Functor.comp_map, comp_app, map_comp, assoc,
+          unit_naturality_assoc, right_triangle_components_assoc]
+        rw [← assoc, ← Functor.comp_map, assoc, ← β.naturality, ← assoc, Functor.comp_map,
+          ← G.map_comp, right_triangle_components, map_id, id_comp]
+#align category_theory.mates CategoryTheory.Mates
+
+/-- A component of a transposed version of the mates correspondence. -/
+theorem Mates_counit (α : G ⋙ L₂ ⟶ L₁ ⋙ H) (d : D) :
```

Comment:
```suggestion
theorem mateEquiv_counit (α : G ⋙ L₂ ⟶ L₁ ⋙ H) (d : D) :
```

Could you consider including similar (dual) lemmas like `(co)unit_mateEquiv_symm`?

Reply:
Done.